src/blob/index/bptree/core.rs

```diff
@@ -62,10 +62,11 @@ where
         let file = File::create(path, ioring)
             .await
             .with_context(|| format!("file open failed {:?}", path))?;
-        file.write_append(&buf).await?;
+        file.write_append_all(buf.freeze()).await?;
         header.set_written(true);
-        let serialized_header = serialize(&header)?;
-        file.write_at(0, &serialized_header).await?;
+        let mut serialized_header = BytesMut::new();
```

Comment:
Same here: It would be good to estimate expected size and create buffer with capacity. Bincode has `serialized_size` function for that

Reply:
You can add `serialized_size` function to Index header struct like it is done for record header